Makefile

```diff
@@ -73,17 +73,19 @@ bin/buildah: $(SOURCES) cmd/buildah/*.go internal/mkcw/embed/entrypoint.gz
 	$(GO_BUILD) $(BUILDAH_LDFLAGS) $(GO_GCFLAGS) "$(GOGCFLAGS)" -o $@ $(BUILDFLAGS) ./cmd/buildah
 
 ifneq ($(shell as --version | grep x86_64),)
+internal/mkcw/embed/entrypoint.gz: internal/mkcw/embed/entrypoint
+	$(RM) $@
+	gzip -k $^
+
 internal/mkcw/embed/entrypoint: internal/mkcw/embed/entrypoint.s
 	$(AS) -o $(patsubst %.s,%.o,$^) $^
 	$(LD) -o $@ $(patsubst %.s,%.o,$^)
 	strip $@
 else
 .PHONY: internal/mkcw/embed/entrypoint
+.PHONY: internal/mkcw/embed/entrypoint.gz
```

Comment:
I confirm this works, but, also, the GNU Make manual says

> A phony target should not be a prerequisite of a real target file; if it is, its recipe will be run every time `make` considers that file. 

AFAICS removing the `else` case entirely would work (or replacing it with a comment pointing at some validation?), and avoid the unnecessary rebuilds.

Reply:
Yes that works too, I've amended the patch and removed both targets.